src/main/java/sirius/web/controller/ControllerDispatcher.java

```diff
@@ -329,17 +336,20 @@ private void handleFailure(WebContext webContext, Route route, Throwable cause)
         }
     }
 
-    /*
+    /**
      * Compiles all available controllers and their methods into a route table
+     *
+     * @return a list holding all available {@link Route routes}
      */
-    private List<Route> buildRouter() {
+    private synchronized List<Route> buildRouter() {
         PriorityCollector<Route> collector = PriorityCollector.create();
         for (Controller controller : Injector.context().getParts(Controller.class)) {
             compileController(collector, controller);
         }
 
         List<Route> allRoutes = collector.getData();
         optimizeRoutes(allRoutes);
+        allRoutes.forEach(route -> publicServices.recordPublicService(route.getMethod()));
```

Comment:
shouldn't we reverse this and move the whole logik to PublicServices an perform this lazy initialization there? This would get rid of all sync blocks here...